Sources/Extensions/SwiftStdlib/CollectionExtensions.swift

```diff
@@ -35,6 +35,18 @@ public extension Collection {
 		}
 	}
 	
+    /// SwifterSwift: Group the elements of the collection in a dictionary.
+    ///
+    ///     [Pair(1, "Hello"), ...].group(by: \Pair.name) -> ["Hello": [Pair(1, "Hello"), ...], ...]
+    ///
+    /// - Parameter key: The `KeyPath` for the element's property to group the collection by.
+    /// - Returns: A dictionary with values grouped with keys.
+    public func group<Value>(by key: KeyPath<Self.Element, Value>) -> [Value: [Self.Element]] {
```

Comment:
This seems pretty inconsistent with the existing extension on Array. Maybe we can try and formalize these two? Not sure if the Array extension can be moved to Collection.